olx_parser.py

```diff
@@ -1,124 +1,186 @@
-            return datetime.strptime(date_of_last_visit, '%d, %B, %Y')
-        elif len(date_time) == 3:
-            day = str(datetime.now().day)
-            month = str(datetime.now().month)
-            year = str(datetime.now().year)
-            time = date_time[2]
-            date_of_last_visit = f'{day}, {month}, {year} {time}'
-            return datetime.strptime(date_of_last_visit, '%d, %m, %Y %H:%M')
-        elif len(date_time) == 4:
-            day = str((datetime.now() - timedelta(days=1)).day)
-            month = str(datetime.now().month)
-            year = str(datetime.now().year)
-            time = date_time[3]
-            date_of_last_visit = f'{day}, {month}, {year} {time}'
-            return datetime.strptime(date_of_last_visit, '%d, %m, %Y %H:%M')
-
-    elif date_time[0] == type(int):
+    if len(date_time) > 4:
+        day, month, year = date_time[1], translate_month_to_en(date_time[2]), date_time[3]
+        date_of_last_visit = f'{day}, {month}, {year}'
+        return datetime.strptime(date_of_last_visit, '%d, %B, %Y')
+    elif len(date_time) == 3:
+        day = str(datetime.now().day)
+        month = str(datetime.now().month)
+        year = str(datetime.now().year)
+        time = date_time[2]
+        date_of_last_visit = f'{day}, {month}, {year} {time}'
+        return datetime.strptime(date_of_last_visit, '%d, %m, %Y %H:%M')
+    elif len(date_time) == 4:
+        day = str((datetime.now() - timedelta(days=1)).day)
+        month = str(datetime.now().month)
+        year = str(datetime.now().year)
+        time = date_time[3]
+        date_of_last_visit = f'{day}, {month}, {year} {time}'
+        return datetime.strptime(date_of_last_visit, '%d, %m, %Y %H:%M')
+
+
+def parse_published_date(date_time) -> datetime:
+    # Дата публикации объявления
+
+    date_time = date_time.split(' ')
+    day = None
+    month = None
+    year = None
+    published_date = None
+    if int(date_time[0]):
         day = date_time[0]
-        month = date[date_time[1]]
+        month = translate_month_to_en(date_time[1])
         year = date_time[2]
-        date_of_last_visit = f'{day} {month} {year}'
-        return datetime.strptime(date_of_last_visit, '%d %B %Y')
+        published_date = f'{day} {month} {year}'
+        return datetime.strptime(published_date, '%d %B %Y')
     elif date_time[0] == 'Сегодня':
         day = str(datetime.now().day)
         month = str(datetime.now().month)
         year = str(datetime.now().year)
         time = date_time[2]
-        date_of_last_visit = f'{day}, {month}, {year} {time}'
-        return datetime.strptime(date_of_last_visit, '%d, %m, %Y %H:%M')
-    
+        published_date = f'{day}, {month}, {year} {time}'
+        return datetime.strptime(published_date, '%d, %m, %Y %H:%M')
+
 
 if __name__ in '__main__':
     url = 'https://www.olx.kz/zhivotnye/?page='
-    get_product(url)
\ No newline at end of file
+    parser_page(url)
```

Comment:
функция возвращает генератор, чтобы процесс парсинга пошел, по генератору надо итерироваться
```suggestion
    for adt_dict in parser_page(url):
        ...
```